maven-plugin/src/main/java/net/revelc/code/formatter/FormatterMojo.java

```diff
@@ -280,7 +280,7 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             String basedirPath = getBasedirPath();
             for (int i = 0, n = files.size(); i < n; i++) {
                 File file = files.get(i);
-                if (file.exists()) {
+                if (file.exists() && file.canWrite()) {
```

Comment:
@fbyrne I'm going to pull this in a separate pull request to master shortly.  I've enhanced it further so we get a count of files that were tagged read only.  This is certainly better than throwing tons of stack traces and will help developers as well as some scm(s) such as mks integrety require same checkout as you noted.